### PR TITLE
Add a pluggable logging backend

### PR DESCRIPTION
**Pull request by atakanarikan**
This PR adds a setting to the library, making it more pluggable to end users.

We have encountered that the database table became too large for our operations, and decided to log our audit events to ElasticSearch instead. With the changes in this pr, it's possible to change the logging backend to whatever the end user prefers, instead of being forced to create a database table for audit logs.

A simple example for kibana is given in the readme.

### CODE REVIEW

**Comment by jheld** [comment]
Looks great overall. Please rebase and I think we can plan for this in the current release target.

**Comment by atakanarikan** [comment]
@jheld thanks! merged the conflict. lmk if i can do something more for you!

**Comment by jheld** [comment]
@atakanarikan I left some minor (but critical) code change requests.

I ran the tests and found the failures.

Unfortunately I had to run them locally (my fork has Travis CI setup so I would have seen it, but it would have been _after_ the merge).

I'm not sure if I want to try pointing people to put their PRs to my fork, yet.

**Comment by atakanarikan** [comment]
@jheld hmm that's weird. I'll check the tests and fix the code this weekend. thanks!

File: easyaudit/signals/auth_signals.py [hunk 1]
```diff
@@ -32,9 +40,11 @@ def user_login_failed(sender, credentials, **kwargs):
         with transaction.atomic():
             request = get_current_request() # request argument not available in django < 1.11
             user_model = get_user_model()
-            login_event = LoginEvent.objects.create(login_type=LoginEvent.FAILED,
-                                                    username=credentials[user_model.USERNAME_FIELD],
-                                                    remote_ip=request.META[REMOTE_ADDR_HEADER])
+            login_event = audit_logger.login({
+                'login_type': LoginEvent.FAILED,
+                'username': getattr(user, user.USERNAME_FIELD),
```
Comment (jheld) [inline easyaudit/signals/auth_signals.py]: sorry just saw: this needs to be based on `user_model` like before. `user` doesn't exist.

**Comment by jheld** [comment]
OK, I've merged it to master, tagged the commit, made a release not on this project and released `1.2.2b2` to PyPI. This is likely the last change for the `1.2.2` release (unless we get any security changes or bug fixes).

**Pull request merged**